lib/anno_utils.pm

```diff
@@ -0,0 +1,2917 @@
+# -*- perl -*-
+
+package anno_utils;
+
+##########################################################################
+# This module is built to handle the annotation of Assembly
+# (KBaseGenomeAnnotations.Assembly or Genome
+# (KBaseGenomes/KBaseGenomeAnnotations.GenomeAnnotation) object(s)
+# and create a KBaseGenomeAnnotations.Assembly/GenomeAnnotation  object/set
+###########################################################################
+
+use strict;
+use warnings;
+
+use Bio::KBase::kbaseenv;
+use Config::Simple;
+use Data::Dumper qw(Dumper);
+use File::Spec::Functions qw(catfile splitpath);
+use File::Path qw(make_path);
+use File::Copy;
+use Carp qw(croak);
+use Bio::SeqIO;
+use Bio::Perl;
+use Bio::Tools::CodonTable;
+use JSON;
+use Encode qw(encode decode);
+use URI::Encode qw(uri_encode uri_decode);
+use File::Basename;
+use Array::Utils qw(:all);
+use Text::Trim qw(trim);
+use Data::Structure::Util qw( unbless );
+
+use Bio::KBase::Exceptions;
+use Bio::KBase::utilities;
+use Bio::KBase::kbaseenv;
+use GenomeTypeObject;
+
+use Bio::KBase::GenomeAnnotation::GenomeAnnotationImpl;
+use installed_clients::GenomeAnnotationAPIClient;
+use installed_clients::GenomeAnnotationClient;
+use installed_clients::AssemblyUtilClient;
+use installed_clients::GenomeFileUtilClient;
+use installed_clients::WorkspaceClient;
+use installed_clients::KBaseReportClient;
+use installed_clients::kb_SetUtilitiesClient;
+
+
+#######################Begin refactor annotate_process########################
+## helper subs
+sub _get_scientific_name_for_NCBI_taxon {
+    my ($self, $tax_id, $timestamp) = @_;
+    my $url = $self->{_re_url} . "/api/v1/query_results?stored_query=ncbi_fetch_taxon";
+    my $content = encode_json({
+        'ts' => $timestamp,
+        'id'=> $tax_id . '' # make sure tax id is a string
+        });
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('content-type', 'application/json');
+    $req->content($content);
+    my $ua = LWP::UserAgent->new();
+    
+    my $ret = $ua->request($req);
```

Comment:
You can combine all the request stuff into one statement and use `LWP::UserAgent`'s `post` shortcut:

```
my $ua = LWP::UserAgent->new;
my $response = $ua->post(
   $self->{_re_url} . "/api/v1/query_results?stored_query=ncbi_fetch_taxon",
   'Content-Type' => 'application/json',
   Content        => $content,
);
unless ( $response->is_success ) {
   ...
}
```